frontend/online-shop/src/Header.tsx

```diff
@@ -0,0 +1,29 @@
+import { Link } from "react-router-dom";
+import Grid from "@material-ui/core/Grid";
+import HomeIcon from "@material-ui/icons/Home";
+import ShoppingCartIcon from "@material-ui/icons/ShoppingCart";
+import { IconButton } from "@material-ui/core";
+import { useStyles } from "./styles/tableStyles";
+
+const Header = () => (
+  <header>
+    <nav>
+      <Grid container spacing={3}>
+        <Grid item xs={12} sm={6}>
+          <Link to={{ pathname: `/` }}>
+            <IconButton className={useStyles().buttonStyle}>
```

Comment:
You should extract the `useStyles()` in a const above the return because the way you do it you call the useStyles() method on each jsx element. If you extract it you will make the call only once and then retrieve the classes from there.

Reply:
Fixed